fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -1399,66 +1448,73 @@ private static void parseInclusionParameter(Class<?> resourceType, FHIRSearchCon
+                throw SearchExceptionUtil.buildNewInvalidSearchException(
+                        "The join resource type must match the resource type being searched.");
             }
-            if (!searchParm.getType().getValue().equals("reference")) {
-                throw SearchExceptionUtil
-                        .buildNewInvalidSearchException("Inclusion Parameter must be of type 'reference'. "
-                                + "The passed Inclusion Parameter is of type: " + searchParm.getType().getValue());
+
+            // For _revinclude parameter, target resource type, if specified, must match resource type being searched
+            if (SearchConstants.REVINCLUDE.equals(inclusionKeyword) && searchParameterTargetType != null
+                    && !searchParameterTargetType.equals(resourceType.getSimpleName())) {
+                throw SearchExceptionUtil.buildNewInvalidSearchException("The search parameter target type must match the resource type being searched.");
             }
 
-            if (inclusionKeyword.equals(SearchConstants.INCLUDE)) {
-                newInclusionParms =
-                        buildIncludeParameter(resourceType, joinResourceType, searchParm, searchParameterName,
-                                searchParameterTargetType);
-                context.getIncludeParameters().addAll(newInclusionParms);
+            // Ensure that the Inclusion Parameter being parsed is a valid search parameter of type 'reference'.
+            Map<String, SearchParameter> searchParametersMap;
+            if (SearchConstants.WILDCARD.equals(searchParameterName)) {
+                searchParametersMap = getInclusionWildcardSearchParameters(resourceType.getSimpleName(), joinResourceType, searchParameterTargetType, inclusionKeyword);
             } else {
-                newInclusionParm =
-                        buildRevIncludeParameter(resourceType, joinResourceType, searchParm, searchParameterName,
-                                searchParameterTargetType);
-                context.getRevIncludeParameters().add(newInclusionParm);
+                searchParm = getSearchParameter(joinResourceType, searchParameterName);
+                if (searchParm == null) {
+                    String msg = "Undefined Inclusion Parameter: " + inclusionValue;
+                    if (lenient) {
+                        // TODO add this to the list of supplemental warnings?
+                        log.fine(msg);
+                        continue;
+                    } else {
+                        throw SearchExceptionUtil.buildNewInvalidSearchException(msg);
+                    }
+                }
+                if (!searchParm.getType().getValue().equals("reference")) {
+                    throw SearchExceptionUtil.buildNewInvalidSearchException("Inclusion Parameter must be of type 'reference'. "
+                            + "The passed Inclusion Parameter is of type: " + searchParm.getType().getValue());
```

Comment:
single quotes around the value.  it shows better in the logs

Reply:
Fixed.